storage/memory.js

```diff
@@ -0,0 +1,96 @@
+'use strict'
+
+const LRUCache = require('mnemonist/lru-cache')
+const StorageInterface = require('./interface')
+
+const DEFAULT_CACHE_SIZE = 1024
+
+/**
+ * @typedef StorageMemoryOptions
+ * @property {?number} [size=1024]
+ * @property {Logger} log
+ */
+
+class StorageMemory extends StorageInterface {
+  /**
+   * in-memory storage
+   * @param {StorageMemoryOptions} options
+   */
+  constructor (options = {}) {
+    // TODO validate options
+    // logger is mandatory
+    super(options)
+    this.store = new LRUCache(options.size || DEFAULT_CACHE_SIZE)
+    this.references = new Map()
+    this.log = options.log
+  }
+
+  async get (key) {
+    const entry = this.store.get(key)
+    if (entry) {
+      if (entry.expires > Date.now()) {
+        return entry.value
+      }
+      this.store.set(key, undefined)
```

Comment:
I think this should clear up the references?

Reply:
imo it's convenient to use a lazy logic + gc, and sync only when necessary
wdyt?

Reply:
I'd go for a gc solution, because it's imo less expensive than checking references every time, and also it fits better with redis, when an entry is removed by redis itself when expires